src/graph/graph_op.cc

```diff
@@ -653,4 +654,77 @@ DGL_REGISTER_GLOBAL("graph_index._CAPI_DGLMapSubgraphNID")
+  if (ig == NULL) {
+    std::cerr << "The input graph isn't immutable graph" << std::endl;
+    return aten::NewIdArray(0);
+  }
+  // This is a symmetric graph, so in-csr and out-csr are the same.
+  CSRPtr csr = ig->GetInCSR();
+  auto mat = csr->ToCSRMatrix();
+
+  idx_t nvtxs = g->NumVertices();
+  idx_t ncon = 1;        // # balacing constraints.
+  idx_t *xadj = static_cast<idx_t*>(mat.indptr->data);
+  idx_t *adjncy = static_cast<idx_t*>(mat.indices->data);
+  // TODO(zhengda) we could use this to create partitions that balance both #vertices and #edges.
+  idx_t *vwgt = NULL;
+  idx_t nparts = k;
+  IdArray part_arr = aten::NewIdArray(nvtxs);
+  idx_t objval = 0;
+  idx_t *part = static_cast<idx_t*>(part_arr->data);
+  int ret = METIS_PartGraphKway(&nvtxs,      // The number of vertices
+                                &ncon,       // The number of balancing constraints.
+                                xadj,        // indptr
+                                adjncy,      // indices
+                                NULL,        // the weights of the vertices
+                                NULL,        // The size of the vertices for computing
+                                // the total communication volume
+                                NULL,        // The weights of the edges
+                                &nparts,     // The number of partitions.
+                                NULL,        // the desired weight for each partition and constraint
+                                NULL,        // the allowed load imbalance tolerance
+                                NULL,        // the array of options
+                                &objval,      // the edge-cut or the total communication volume of
+                                // the partitioning solution
+                                part);
+  printf("Partition a graph with %ld nodes and %ld edges into %d parts and get %ld edge cuts\n",
+         g->NumVertices(), g->NumEdges(), k, objval);
+  if (ret == METIS_OK) {
+    return part_arr;
+  } else if (ret == METIS_ERROR_INPUT) {
+    std::cerr << "Error in Metis partitioning: input error" << std::endl;
+  } else if (ret == METIS_ERROR_MEMORY) {
+    std::cerr << "Error in Metis partitioning: cannot allocate memory" << std::endl;
+  } else {
+    std::cerr << "Error in Metis partitioning: other errors" << std::endl;
+  }
+  // return an array of 0 elements to indicate the error.
+  return aten::NewIdArray(0);
+}
+
+DGL_REGISTER_GLOBAL("transform._CAPI_DGLMetisPartition")
+.set_body([] (DGLArgs args, DGLRetValue* rv) {
+    GraphRef g = args[0];
+    int k = args[1];
+    *rv = GraphOp::MetisPartition(g.sptr(), k);
+  });
+
+#else
+
+DGL_REGISTER_GLOBAL("transform._CAPI_DGLMetisPartition")
+.set_body([] (DGLArgs args, DGLRetValue* rv) {
+    GraphRef g = args[0];
+    int k = args[1];
+    LOG(WARNING) << "DGL doesn't support METIS partitioning in Windows";
```

Comment:
Let's just throw error. No need to warn.

Reply:
is the exception captured in python or just crash the python process?